### PR TITLE
Viite 458 change roadadress numbering for whole road part

### PR DESCRIPTION
**Pull request by Campanha**

### CODE REVIEW

File: digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/RoadAddressValidator.scala [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  }
+
+  def checkNotReserved(number: Long, part: Long, currentProject: RoadAddressProject) = {
+    val project = ProjectDAO.roadPartReservedByProject(number, part, currentProject.id, withProjectId = true)
+    if (project.nonEmpty) {
+      throw new ProjectValidationException(s"TIE $number OSA $part on jo varattuna projektissa ${project.get}, tarkista tiedot")
+    }
+  }
+
+  def fetchProject(id: Long): RoadAddressProject = {
```
Comment (tvenhola) [inline digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/RoadAddressValidator.scala]: In my opinion this shouldn't be part of validator, but if changed to checkProjectExists(id: Long) and being Unit then yes.

File: digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/RoadAddressValidator.scala [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package fi.liikennevirasto.viite
+
+import fi.liikennevirasto.viite.dao.{ProjectDAO, RoadAddressDAO, RoadAddressProject}
+import org.joda.time.format.DateTimeFormat
+
+object RoadAddressValidator {
+
+  def checkAvailable(number: Long, part: Long, currentProject: RoadAddressProject) = {
```
Comment (tvenhola) [inline digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/RoadAddressValidator.scala]: Force return types, Unit

File: digiroad2-viite/src/test/scala/fi/liikennevirasto/viite/ProjectServiceSpec.scala [hunk 3]
```diff
@@ -448,7 +448,7 @@ class ProjectServiceSpec  extends FunSuite with Matchers with BeforeAndAfter {
       val projectLinks = ProjectDAO.fetchByProjectNewRoadPart(5, 205, saved.id)
       count = countCurrentProjects.size + 1
       countAfterInsertProjects.size should be(count)
-      sqlu"""UPDATE Project_link set status = 1""".execute
+      sqlu"""UPDATE Project_link set status = 5""".execute
```
Comment (tvenhola) [inline digiroad2-viite/src/test/scala/fi/liikennevirasto/viite/ProjectServiceSpec.scala]: Use enumeration here, exactly for this reason
replace 5 with ${LinkStatus.Terminated.value}

File: digiroad2-oracle/src/main/resources/db/migration/V122__update_link_status_in_project_link_according_to_TR.sql [hunk 4]
```diff
@@ -0,0 +1,8 @@
+UPDATE PROJECT_LINK SET STATUS = CASE STATUS WHEN 1 THEN 5
+                                             WHEN 2 THEN 2
+                                             WHEN 3 THEN 3
+                                             WHEN 4 THEN 1
+                                             WHEN 5 THEN 4
+                                             WHEN 0 THEN 0
+                                             ELSE 99
```
Comment (plourenco90) [inline digiroad2-oracle/src/main/resources/db/migration/V122__update_link_status_in_project_link_according_to_TR.sql]: I think here also would be better with the enum types

Comment (tvenhola) [reply digiroad2-oracle/src/main/resources/db/migration/V122__update_link_status_in_project_link_according_to_TR.sql]: Within SQL migration script we don't have access to them.

Comment (plourenco90) [reply digiroad2-oracle/src/main/resources/db/migration/V122__update_link_status_in_project_link_according_to_TR.sql]: Right, didn't noticed it was a migration script, sorry!

**Pull request merged**